### ISSUE TITLE
Unable to install nevergrad on MacOS (M1 chip)

### ISSUE THREAD

**Issue by ArrunAnanda** [facebookexperimental/Robyn#407]
Hello team,

I'm currently using MacOS and I'm unable to install "Nevergrad". I get this below error. Can you please help me with this?

I'm using the Option 1 way of installing nevergrad via PIP

----

   ERROR: Command errored out with exit status 1:
     command: /Users/arunananda/.virtualenvs/r-reticulate/bin/python /Users/arunananda/.virtualenvs/r-reticulate/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/tmp93cubnaw
         cwd: /private/var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/pip-install-8h6h2vji/scipy
    Complete output (218 lines):
    Running scipy/linalg/_generate_pyx.py
    Running scipy/special/_generate_pyx.py
    Running scipy/stats/_generate_pyx.py
    Processing scipy/cluster/_vq.pyx
    Processing scipy/cluster/_optimal_leaf_ordering.pyx
    Processing scipy/cluster/_hierarchy.pyx
    Processing scipy/ndimage/src/_cytest.pyx
    Processing scipy/ndimage/src/_ni_label.pyx
    Processing scipy/linalg/cython_lapack.pyx
    Processing scipy/linalg/_matfuncs_sqrtm_triu.pyx
    Processing scipy/linalg/_cythonized_array_utils.pyx
    Processing scipy/linalg/_solve_toeplitz.pyx
    Processing scipy/linalg/cython_blas.pyx
    Processing scipy/linalg/_decomp_update.pyx.in
    Processing scipy/optimize/_bglu_dense.pyx
    Processing scipy/optimize/_group_columns.pyx
    Processing scipy/optimize/_trlib/_trlib.pyx
    Processing scipy/optimize/_highs/cython/src/_highs_constants.pyx
    Processing scipy/optimize/_highs/cython/src/_highs_wrapper.pyx
    Processing scipy/optimize/_lsq/givens_elimination.pyx
    Processing scipy/optimize/cython_optimize/_zeros.pyx.in
    Processing scipy/optimize/tnc/_moduleTNC.pyx
    Processing scipy/io/matlab/_streams.pyx
    Processing scipy/io/matlab/_mio_utils.pyx
    Processing scipy/io/matlab/_mio5_utils.pyx
    Processing scipy/_lib/_ccallback_c.pyx
    Processing scipy/_lib/_test_deprecation_def.pyx
    Processing scipy/_lib/_test_deprecation_call.pyx
    Processing scipy/_lib/messagestream.pyx
    Processing scipy/special/_ufuncs_cxx.pyx
    Processing scipy/special/cython_special.pyx
    Processing scipy/special/_ellip_harm_2.pyx
    Processing scipy/special/_comb.pyx
    Processing scipy/special/_test_round.pyx
    Processing scipy/special/_ufuncs.pyx
    Processing scipy/fftpack/convolve.pyx
    Processing scipy/interpolate/interpnd.pyx
    Processing scipy/interpolate/_bspl.pyx
    Processing scipy/interpolate/_ppoly.pyx
    Processing scipy/sparse/_csparsetools.pyx.in
    Processing scipy/sparse/csgraph/_shortest_path.pyx
    Processing scipy/sparse/csgraph/_traversal.pyx
    Processing scipy/sparse/csgraph/_flow.pyx
    Processing scipy/sparse/csgraph/_tools.pyx
    Processing scipy/sparse/csgraph/_matching.pyx
    Processing scipy/sparse/csgraph/_reordering.pyx
    Processing scipy/sparse/csgraph/_min_spanning_tree.pyx
    Processing scipy/spatial/_qhull.pyx
    Processing scipy/spatial/_ckdtree.pyx
    Processing scipy/spatial/_voronoi.pyx
    Processing scipy/spatial/_hausdorff.pyx
    Processing scipy/spatial/transform/_rotation.pyx
    Processing scipy/signal/_max_len_seq_inner.pyx
    Processing scipy/signal/_peak_finding_utils.pyx
    Processing scipy/signal/_upfirdn_apply.pyx
    Processing scipy/signal/_spectral.pyx
    Processing scipy/signal/_sosfilt.pyx
    Processing scipy/stats/_stats.pyx
    Processing scipy/stats/_qmc_cy.pyx
    Processing scipy/stats/_biasedurn.pyx
    Processing scipy/stats/_sobol.pyx
    Processing scipy/stats/_boost/src/beta_ufunc.pyx
    Processing scipy/stats/_boost/src/nbinom_ufunc.pyx
    Processing scipy/stats/_boost/src/hypergeom_ufunc.pyx
    Processing scipy/stats/_boost/src/binom_ufunc.pyx
    Processing scipy/stats/_unuran/unuran_wrapper.pyx
    warning: unuran_wrapper.pyx:470:21: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:470:28: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:470:36: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:515:21: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:515:28: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:515:36: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:1469:21: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:1469:28: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    warning: unuran_wrapper.pyx:1469:36: Non-trivial type declarators in shared declaration (e.g. mix of pointers and values). Each pointer declaration should be on its own line.
    Cythonizing sources
    lapack_opt_info:
    lapack_mkl_info:
    customize UnixCCompiler
      libraries mkl_rt not found in ['/Users/arunananda/.virtualenvs/r-reticulate/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    openblas_lapack_info:
    customize UnixCCompiler
    customize UnixCCompiler
      libraries openblas not found in ['/Users/arunananda/.virtualenvs/r-reticulate/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    openblas_clapack_info:
    customize UnixCCompiler
    customize UnixCCompiler
      libraries openblas,lapack not found in ['/Users/arunananda/.virtualenvs/r-reticulate/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    flame_info:
    customize UnixCCompiler
      libraries flame not found in ['/Users/arunananda/.virtualenvs/r-reticulate/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    atlas_3_10_threads_info:
    Setting PTATLAS=ATLAS
    customize UnixCCompiler
      libraries lapack_atlas not found in /Users/arunananda/.virtualenvs/r-reticulate/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /Users/arunananda/.virtualenvs/r-reticulate/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries tatlas,tatlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_3_10_threads_info'>
      NOT AVAILABLE
    
    atlas_3_10_info:
    customize UnixCCompiler
      libraries lapack_atlas not found in /Users/arunananda/.virtualenvs/r-reticulate/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /Users/arunananda/.virtualenvs/r-reticulate/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries satlas,satlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_3_10_info'>
      NOT AVAILABLE
    
    atlas_threads_info:
    Setting PTATLAS=ATLAS
    customize UnixCCompiler
      libraries lapack_atlas not found in /Users/arunananda/.virtualenvs/r-reticulate/lib
    customize UnixCCompiler
      libraries ptf77blas,ptcblas,atlas not found in /Users/arunananda/.virtualenvs/r-reticulate/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries ptf77blas,ptcblas,atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries ptf77blas,ptcblas,atlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_threads_info'>
      NOT AVAILABLE
    
    atlas_info:
    customize UnixCCompiler
      libraries lapack_atlas not found in /Users/arunananda/.virtualenvs/r-reticulate/lib
    customize UnixCCompiler
      libraries f77blas,cblas,atlas not found in /Users/arunananda/.virtualenvs/r-reticulate/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries f77blas,cblas,atlas not found in /usr/local/lib
    customize UnixCCompiler
      libraries lapack_atlas not found in /usr/lib
    customize UnixCCompiler
      libraries f77blas,cblas,atlas not found in /usr/lib
    <class 'numpy.distutils.system_info.atlas_info'>
      NOT AVAILABLE
    
    accelerate_info:
      NOT AVAILABLE
    
    lapack_info:
    customize UnixCCompiler
      libraries lapack not found in ['/Users/arunananda/.virtualenvs/r-reticulate/lib', '/usr/local/lib', '/usr/lib']
      NOT AVAILABLE
    
    lapack_src_info:
      NOT AVAILABLE
    
      NOT AVAILABLE
    
    setup.py:486: UserWarning: Unrecognized setuptools command ('dist_info --egg-base /private/var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/pip-modern-metadata-g3z9sybx'), proceeding with generating Cython sources and expanding templates
      warnings.warn("Unrecognized setuptools command ('{}'), proceeding with "
    Running from SciPy source directory.
    /private/var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/pip-build-env-xgz126jn/overlay/lib/python3.8/site-packages/numpy/distutils/system_info.py:1712: UserWarning:
        Lapack (http://www.netlib.org/lapack/) libraries not found.
        Directories to search for the libraries can be specified in the
        numpy/distutils/site.cfg file (section [lapack]) or by setting
        the LAPACK environment variable.
      if getattr(self, '_calc_info_{}'.format(lapack))():
    /private/var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/pip-build-env-xgz126jn/overlay/lib/python3.8/site-packages/numpy/distutils/system_info.py:1712: UserWarning:
        Lapack (http://www.netlib.org/lapack/) sources not found.
        Directories to search for the sources can be specified in the
        numpy/distutils/site.cfg file (section [lapack_src]) or by setting
        the LAPACK_SRC environment variable.
      if getattr(self, '_calc_info_{}'.format(lapack))():
    Traceback (most recent call last):
      File "/Users/arunananda/.virtualenvs/r-reticulate/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
        main()
      File "/Users/arunananda/.virtualenvs/r-reticulate/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/Users/arunananda/.virtualenvs/r-reticulate/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 133, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/private/var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/pip-build-env-xgz126jn/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 174, in prepare_metadata_for_build_wheel
        self.run_setup()
      File "/private/var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/pip-build-env-xgz126jn/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 267, in run_setup
        super(_BuildMetaLegacyBackend,
      File "/private/var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/pip-build-env-xgz126jn/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 158, in run_setup
        exec(compile(code, __file__, 'exec'), locals())
      File "setup.py", line 628, in <module>
        setup_package()
      File "setup.py", line 624, in setup_package
        setup(**metadata)
      File "/private/var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/pip-build-env-xgz126jn/overlay/lib/python3.8/site-packages/numpy/distutils/core.py", line 137, in setup
        config = configuration()
      File "setup.py", line 526, in configuration
        raise NotFoundError(msg)
    numpy.distutils.system_info.NotFoundError: No BLAS/LAPACK libraries found. Note: Accelerate is no longer supported.
    To build Scipy from sources, BLAS & LAPACK libraries need to be installed.
    See site.cfg.example in the Scipy source directory and
    https://docs.scipy.org/doc/scipy/reference/building/index.html for details.
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/arunananda/.virtualenvs/r-reticulate/bin/python /Users/arunananda/.virtualenvs/r-reticulate/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /var/folders/v9/nlhgzkv97hlb6kywp4yp6v9h0000gp/T/tmp93cubnaw Check the logs for full command output.
WARNING: You are using pip version 20.2.3; however, version 22.1.2 is available.
You should consider upgrading via the '/Users/arunananda/.virtualenvs/r-reticulate/bin/python -m pip install --upgrade pip' command.
Error: Error installing package(s): "'nevergrad'"

**Comment by ArrunAnanda** [comment]
<img width="1563" alt="image" src="https://user-images.githubusercontent.com/59400103/173030012-07e6e449-91eb-408d-8687-54e401e16662.png">

**Comment by ArrunAnanda** [comment]
Hi, 

I managed to make it work. 

In short - 

1. What I noticed was Python wasn't installed even though it shows up in the virtualenvs file. 
2. Then I installed miniconda as suggested in the documentation and it worked. 

I'm attaching the screenshots for further reference. 

Thanks!

<img width="1440" alt="3" src="https://user-images.githubusercontent.com/59400103/173843911-90960573-e851-47fa-bbb2-30cb1f5f7b93.png">

<img width="1440" alt="1" src="https://user-images.githubusercontent.com/59400103/173843896-dad37a11-469b-49a4-841c-efa0b623e994.png">
<img width="1440" alt="2" src="https://user-images.githubusercontent.com/59400103/173843906-cebf6f8f-1e51-46dd-aafa-a2182eb7b379.png">
